app/controllers/repos_controller.rb

```diff
@@ -26,10 +26,16 @@ def show
 
       @charts = @benchmark_type.benchmark_result_types.map do |benchmark_result_type|
         cache_key = "#{BenchmarkRun.charts_cache_key(@benchmark_type, benchmark_result_type)}:#{@benchmark_run_display_count}"
+        version_cache_key = "version_#{cache_key}"
 
-        if (columns = $redis.get(cache_key))
+        if (columns = $redis.get(cache_key)) && (versions = $redis.get(version_cache_key))
+          # read cached @versions
+          @versions = JSON.parse(versions)
```

Comment:
The line above (`@benchmark_type.benchmark_result_types.map`) tells us that this block of code will be executed multiple times. Then `@versions = JSON.parse(versions)` will also run multiple times, overriding the previous values every time. Perhaps we need a different structure?